### DISCUSSION TITLE
Has Anybody Created a Portainer.io Application Template for Budibase?

### DISCUSSION THREAD

**Discussion by tinjaw** [Budibase/budibase#2275] [Help] [answered]
I would be interested in cloning it.

If nobody has, I guess I will, and make it available. In that case, anybody else a Portainer user that could collaborate with me?

**Comment by shogunpurple** [comment]
Hi @tinjaw - I haven't heard of anyone creating one so far, but feel free to do so and if you have any questions let us know and we can help you with the application template JSON. 

We are quite big fans of portainer, it would be cool to have this!

**Comment by aptkingston** [comment]
Hey @tinjaw. I tried (briefly) at one stage to get it to run in Portainer and didn't manage to, because we can't define our stack in a single docker compose file. We have the `envoy.yaml` config file on disk which was always a problem. Our other file on disk is just our environment variables which can be defined in portainer so that's not a problem. I just couldn't get it working because of the dependency we have on an external config file - but I would love to run it in Portainer if you have an idea of how to get it to work!

**Reply by tinjaw** [reply]
I will try to work with the Portainer folks on this. I need to learn more about compose files and Portainer, so this will serve as my learning oppertunity.

**Comment by Bogey** [comment]
I created a stack in portainer from the BB docker-compose file with minimal adjustments.
I haven’t looked into templates yet. Just tell me how I can help.

**Reply by tinjaw** [reply]
I am going to tackle this now. Can you please provide me with what you have working? I will try to make it a template and submit it for testing with the community.

**Reply by Bogey** [reply]
Easy, go to the BudiBase repository, get "docker-compose.yaml" from the "hosting" folder.
Get from the same folder  the file "hosting.properties".
Open "hosting.properties" and edit the environment settings to your situation/preference.
Save the file as "budibase.env".
 
Open Portainer, go to 'Stacks', copy-paste the contents from "docker-compose.yaml" in the 'web-editor textarea'.
Scroll down, click 'Load variables from .env file' and select the local file "budibase.env".
Click 'Deploy stack'.
Done.

**Reply by SlayterDev** [reply]
@Bogey how did you get around the need for the `envoy.yml` file? Following your instructions the envoy container will not start and it appears to be an error with needing this file.

**Reply by Bogey** [reply]
I didn't, I just used the exact file that's in the github repo. Point the volume to where your file is located:
```
volumes:
      - /your/local/path/envoy.yaml:/etc/envoy/envoy.yaml
```
As for the other volumes in the docker-compose file, you might want to adjust the paths so it reflects your configuration.

**Reply by SlayterDev** [reply]
So you downloaded the repo before deploying your stack then? I just had portainer pull the image which may be part of the issue.

**Reply by Bogey** [reply]
I just downloaded the three files from the [hosting folder](https://github.com/Budibase/budibase/tree/develop/hosting):

- docker-compose.yaml
- envoy.yaml
- hosting.properties (I renamed this to budibase.env so portainer could import it)

I changed the volume paths in docker-compose.yaml, copied the content of the file to portainer stack editor. Changed the contents (passwords) in hosting.properies, renamed it and uploaded it to portainer (or you could copy paste the environment variables in advanced mode). Finally I put envoy.yaml where docker can find it (meaning it’s where you said it would be in the volume definition). That’s it, ready to run.

**Comment by Tarantulino-G** [comment]
> 

I managed to make it work without having to pass the envoy.yaml to the server.

Because of it I had to create a config in portainer, which I pointed to in the stack yaml.
Also I created a portainer volume for logs and pointed to it in the stack yaml.
Also set version to 3.3.

This works but:
-no https and I have to work on getting that working
-I'm not sure I can pass a config file in a portainer template
-having a portainer volume for logs is bad probably (???) so if you already have a ./logs directory in your system it would be better

Let me know if you're interested in this crippled template

P.S.: I use docker in swarm mode so some thing may not be compatible

**Reply by sudipmandal** [reply]
@IvanovSs Yes Pls!

**Comment by aptkingston** [comment, marked as answer]
Quick update about this one - I've added a portainer template to our repository! For now it's in the `develop` branch, so if you'd like to try it out then go to your portainer settings, and add
`https://raw.githubusercontent.com/Budibase/budibase/develop/hosting/portainer/template.json`
to the field for app templates URL:
![image](https://user-images.githubusercontent.com/9075550/161999935-706a9de7-7e83-4e5b-8eee-3311274959d8.png)

Then go to your templates section and you'll see Budibase!
![image](https://user-images.githubusercontent.com/9075550/162000055-efba21ef-ef4f-43f8-9b81-69543ddb1ec3.png)

As for the settings - choose whatever you want for the main port (or leave it at the default) and then just enter some secure random values for the other secrets:
![image](https://user-images.githubusercontent.com/9075550/162000329-93bc1a93-d06f-49e9-b58d-d0076a6f46be.png)

The first time you deploy it will probably take a few minutes as it pulls down all the images. Then just visit your host on whatever port you picked and enjoy your new Budibase installation!
